packages/yoastseo/src/scoring/assessments/seo/TextLengthAssessment.js

```diff
@@ -52,15 +53,53 @@ export default class TextLengthAssessment extends Assessment {
 	 */
 	getResult( paper, researcher ) {
 		const wordCount = researcher.getResearch( "wordCountInText" );
-		const assessmentResult = new AssessmentResult();
+
+		if	( researcher.getConfig( "textLength" ) ) {
+			this._config = this.getLanguageSpecificConfig( researcher );
+		}
 		const calculatedResult = this.calculateResult( wordCount );
 
+		const assessmentResult = new AssessmentResult();
 		assessmentResult.setScore( calculatedResult.score );
 		assessmentResult.setText( calculatedResult.resultText );
 
 		return assessmentResult;
 	}
 
+	/**
+	 * Check if there is language-specific config, and if so, overwrite the current config with it.
+	 *
+	 * @param {Researcher} researcher The researcher to use.
+	 *
+	 * @returns {Object} The config that should be used.
+	 */
+	getLanguageSpecificConfig( researcher ) {
+		const currentConfig = this._config;
+		const languageSpecificConfig = researcher.getConfig( "textLength" );
+
+		// Don't do anything if there is no language-specific config.
+		if ( ! languageSpecificConfig ) {
```

Comment:
I think this will never be true because this method is only called when the config exists. But if you want to keep it, I'd (try to) add a spec for it because currently this line is uncovered by specs.

Reply:
Good point, I've removed the check 👍